tests/farmer_harvester/test_farmer.py

```diff
@@ -0,0 +1,519 @@
+            IncrementPoolStatsCase(
+                std_hash(b"0"),
+                "xxx",
+                1689491043.3493967,
+                1,
+                1,
+                None,
+            ),
+            id="p2_singleton_puzzle_hash_not_exist",
+        ),
+        pytest.param(
+            IncrementPoolStatsCase(
+                std_hash(b"11223344"),
+                "yyy",
+                1689491043.3493967,
+                1,
+                1,
+                {
+                    "p2_singleton_puzzle_hash": std_hash(b"11223344").hex(),
+                    "xxx_since_start": 1,
+                    "xxx_24h": [(1689491043, 1)],
+                    "current_difficulty": 1,
+                },
+            ),
+            id="p2_singleton_puzzle_hash_exist_but_no_change",
+        ),
+        pytest.param(
+            IncrementPoolStatsCase(
+                std_hash(b"11223344"),
+                "xxx",
+                1689491044.1,
+                1,
+                2,
+                {
+                    "p2_singleton_puzzle_hash": std_hash(b"11223344").hex(),
+                    "xxx_since_start": 2,
+                    "xxx_24h": [(1689491043, 1), (1689491044, 2)],
+                    "current_difficulty": 1,
+                },
+            ),
+            id="value_is_set",
+        ),
+        pytest.param(
+            IncrementPoolStatsCase(
+                std_hash(b"11223344"),
+                "xxx",
+                1689491044.1,
+                1,
+                None,
+                {
+                    "p2_singleton_puzzle_hash": std_hash(b"11223344").hex(),
+                    "xxx_since_start": 2,
+                    "xxx_24h": [(1689491043, 1), (1689491044, 1)],
+                    "current_difficulty": 1,
+                },
+            ),
+            id="value_is_not_set",
+        ),
+        pytest.param(
+            IncrementPoolStatsCase(
+                std_hash(b"11223344"),
+                "xxx",
+                1689577444,
+                1,
+                None,
+                {
+                    "p2_singleton_puzzle_hash": std_hash(b"11223344").hex(),
+                    "xxx_since_start": 2,
+                    "xxx_24h": [(1689577444, 1)],
+                    "current_difficulty": 1,
+                },
+            ),
+            id="stripped",
+        ),
+    ],
+)
+def test_increment_pool_stats(case: IncrementPoolStatsCase):
+    increment_pool_stats(
+        case.pool_states, case.p2_singleton_puzzle_hash, case.name, case.current_time, case.count, case.value
+    )
+    if case.expected_result is None:
+        assert case.p2_singleton_puzzle_hash not in case.pool_states
+    else:
+        assert case.pool_states[case.p2_singleton_puzzle_hash] == case.expected_result
+
+
+@pytest.mark.parametrize(
+    argnames="case",
+    argvalues=[
+        pytest.param(
+            NewProofOfSpaceCase.create_verified_quality_case(
```

Comment:
I think you should create a helper function to construct `NewProofOfSpaceCase` where you can omit some values (in these cases, you omit most) and just leave in the ones relevant for the test. But, given that so many of these parameters stay the same, do you really need to parameterize the test by them?

Reply:
Resolving comments in test code - we should iterate and simplfy post release